server/src/main/java/org/opensearch/extensions/rest/AuthorizationRequest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.extensions.rest;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.transport.TransportRequest;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Request to evaluate user privileges for extension actions
+ *
+ * @opensearch.internal
+ */
+public class AuthorizationRequest extends TransportRequest {
+
+    private String extensionUniqueId;
+    // TODO Switch to principal identifier object that is passed to extension
+    private String requestIssuerIdentity;
+    private String permissionId;
+    private Map<String, Object> params;
```

Comment:
Object seems too vague allowing callers to insert all kinds of values, is there a set of values that would make sense?

Reply:
I believe we want anything Writeable to StreamOutput, in general that would be classes implementing Writeable and primitives.

Reply:
Since these are the _checkable_ params, I will starting creating a class called `Checkable` that is writeable and experiment using it with extensions.

Reply:
Changed this to WriteableParameter/CheckableParameter which are added in this PR. 

For questions like `Can I write a document on index X?`, then we can supply the parameters needed to fully determine the authorization in this data structure.